src/ansi-c/c_typecheck_expr.cpp

```diff
@@ -487,17 +471,10 @@ void c_typecheck_baset::typecheck_expr_main(exprt &expr)
 
 void c_typecheck_baset::typecheck_expr_comma(exprt &expr)
 {
-  if(expr.operands().size()!=2)
-  {
-    error().source_location = expr.source_location();
-    error() << "comma operator expects two operands" << eom;
-    throw 0;
-  }
-
-  expr.type()=expr.op1().type();
+  expr.type() = to_binary_expr(expr).op1().type();
```

Comment:
This loses an error message? If we have established this before, should the method actually take a `binary_exprt`?

Reply:
I did consider that. However, note that we are in a front-end. In particular, it's not guaranteed that all class invariants of `index_exprt` are already established at this point. We only rely on the expression having two operands.

Reply:
Possibly misplaced comment? Ack on the `index_exprt` part, I've resolved that earlier comment of mine. Here, however, I still think that the method should take a `binary_exprt`.

Reply:
In principle agreed on the method signature -- but I would say that this needs a separate PR, with a more principled approach.